libs/vkd3d/command.c

```diff
@@ -6403,6 +6414,35 @@ static void STDMETHODCALLTYPE d3d12_command_list_CopyResource(d3d12_command_list
     VKD3D_BREADCRUMB_COMMAND(COPY);
 }
 
+static void d3d12_command_list_end_batch(struct d3d12_command_list *list)
+{
+    switch (list->transfer_batch.batch_type)
+    {
+        case VKD3D_BATCH_TYPE_COPY_BUFFER_TO_IMAGE:
+        case VKD3D_BATCH_TYPE_COPY_IMAGE_TO_BUFFER:
+        case VKD3D_BATCH_TYPE_COPY_IMAGE:
+            d3d12_command_list_end_current_render_pass(list, false);
+            for (size_t i = 0; i < list->transfer_batch.batch_len; i++)
```

Comment:
I'm not sure it's safe to do this commit without also do the alias checks. If there are overlapped inputs, they will conflict I think, performing the same layout transition twice.

Reply:
I've reorganized the commits so the first one introduces a batch without transposing the loop, and the second adds all the alias checks and transpose.